src/Form.js

```diff
@@ -54,13 +54,27 @@ export class Form extends Component {
                 return;
             }
 
-            wrappedChildren.push(React.cloneElement(child, {
-                key: child.props.fieldKey ? child.props.fieldKey : child.type+i,
-                fieldRef : child.ref,
-                ref: child.ref,
+            const isTestable = "hasTestableWrappers" in this.props && this.props.hasTestableWrappers === true;
+            
+            let formElement = isTestable ? child.props.children : child; 
+
+            formElement = React.cloneElement(formElement, {
+                key: formElement.props.fieldKey ? formElement.props.fieldKey : formElement.type+i,
+                fieldRef : formElement.ref,
+                ref: formElement.ref,
                 onFocus:this.handleFieldFocused.bind(this),
-                onChange:this.handleFieldChange.bind(this, child.ref)
-            }));
+                onChange:this.handleFieldChange.bind(this, formElement.ref)
+            });
+        
+            if (isTestable){
+                child = React.cloneElement(child, {
```

Comment:
> Will `child` always be a container?
If the `hasTestableWrappers` prop has been set it would require all of the children to be wrapped in a View of some kind. I see the point you made in the call of essentially what if some `formElements` have unintentionally not been wrapped in a `TView` and would happily add some check to ensure the wrappers are there if the `hasTestableWrappers` prop is set, but I don't know what that check would be. I could add some check to see if the child has the `tid` or `testPathSegment` prop and that one of the props is set, but I think that could lead to confusion and a more difficult issue to pin down if the formElements themselves have their own test IDs. For example with a DisplayText element in flows, formElement would be a TText element.

> Are there any technical risks in adding new "wrapping" views?
I am unaware of any. I've added @chukcha-wtf for review to double check, etc

Reply:
@scottpage1317 has it been tested with AxsyForm and you're sure that values was correctly collected automatically?

I guess the reason why there were this line:
```
React.cloneElement(formElement, {
    key:  ...
    ...
}
```
was due to the fact that this library's `Form` component was trying to assign keys for child objects so it can later automagically collect all values by component keys. Whereas if we'll wrap it into another `React.cloneElement` we might actually loose this ability.

But if I'm wrong and this doesn't really affect the data collection and AxsyForm updates - then the solution looks feasible to me.

Reply:
I've tested it using a Flow in the app which contains all the screen components on the first screen and shows all the values entered on the second screen. That works fine with and without the `hasTestableWrappers` prop.